src/tools/sign-oci-layer-root-hashes/src/sign.rs

```diff
@@ -36,6 +50,137 @@ struct LayerInfo {
     signature: String,
 }
 
+pub(super) async fn attach_signatures(
+    image: &ImageInfo
+) -> Result<(), Error> {
+    let image_ref: Reference = image.name.to_string().parse().unwrap();
+    let client = Client::new(ClientConfig {
+            platform_resolver: Some(Box::new(linux_amd64_resolver)),
+            ..Default::default()
+        });
+    // Authenticate with the registry (if needed)
+    let auth: RegistryAuth = registry::build_auth(&image_ref);
+    let op = RegistryOperation::Push;
+    client.auth(&image_ref, &auth, op).await?;
+
+    let accepted_media_types = &[
+        oci_client::manifest::OCI_IMAGE_INDEX_MEDIA_TYPE,
+        oci_client::manifest::OCI_IMAGE_MEDIA_TYPE,
+        oci_client::manifest::IMAGE_MANIFEST_LIST_MEDIA_TYPE,
+        oci_client::manifest::IMAGE_MANIFEST_MEDIA_TYPE,
+    ];
+    // 1. Resolve the subject descriptor
+    let (raw_bytes, subject_digest) = client
+        .pull_manifest_raw(&image_ref, &auth, accepted_media_types)
+        .await?;
+    // Parse the raw_bytes into an OCI manifest
+    let image_manifest: OciManifest = serde_json::from_slice(&raw_bytes)
+        .context("Failed to parse raw bytes into OCI manifest")?;
+    // Print the digest, raw bytes length, the manifest media type
+    println!("Subject Digest: {}", subject_digest);
+    println!("Raw bytes length: {}", raw_bytes.len());
+
+    let subject_descriptor = OciDescriptor {
+        digest: subject_digest, // Digest of the manifest to attach
+        size: raw_bytes.len() as i64, // Size in bytes
+        media_type: image_manifest.content_type().to_string(),
+        ..Default::default()
+    };
+
+    // 2. Prepare the signature blobs to be attached
```

Comment:
Perhaps add a comment that this is just an initial implementation and the plan is to move later to per layer signature manifest (unless you disagree).

Reply:
Looking at the TS PR, perhaps I misunderstood the implementation a bit here, but we should also highlight that I think we would want to keep the signature blobs separate from the manifest eventually.

Reply:
Added the comment for the layout of the signature manifest. Now only one signature manifest is generated for signing at a time. The signature manifest contains a list of signature blobs for each layer in the image. This will make the snapshotter easier to pick the signatures signed by different signers in multiple signatures scenario. Also append the serial number of the signer certificate to the tag for same purpose.